staging/src/k8s.io/dynamic-resource-allocation/structured/pools.go

```diff
@@ -65,6 +66,24 @@ func GatherPools(ctx context.Context, slices []*resourceapi.ResourceSlice, node
 				return nil, fmt.Errorf("failed to perform node selection for slice %s: %w", slice.Name, err)
 			}
 			if match {
+				if hasBindingConditions(slice) {
+					needBindingSlice := slice.DeepCopy()
+					needBindingSlice.Spec.Devices = nil
+					nodeSlice := slice.DeepCopy()
+					nodeSlice.Spec.Devices = nil
+					for _, device := range slice.Spec.Devices {
```

Comment:
This splits devices from a single ResourceSlice into two separate slices with the same name. I have two concerns about this:
* I don't think this will work with the partitionable devices feature, since it relies on managing the consumption of counters across devices within a ResourceSlice. This duplicates the counters, which will almost certainly lead to oversubscription of the underlying devices.
* We end up with two ResourceSlice objects with the same name, which means that lookup of a specific ResourceSlice by name will not work. Have we checked whether there are any logic today that relies on this? In any case, it seems like it would be surprising behavior for anyone adding features in the future.

Reply:
Also, doesn't the next `case` statement that covers `perDeviceNodeSelection` also need to handle this?

Reply:
As far as I understand, this operation does not split a single ResourceSlice into two. They are then combined again into a single ResourceSlice by the `addSlice()` operation. In this sequence, we are just switching the order.
If my understanding is incorrect, I would appreciate it if you could let me know.

Reply:
> > Also, doesn't the next case statement that covers perDeviceNodeSelection also need to handle this?
> 
> I'll check if it's necessary in this case.

I think this should be added to this case too. I will add it.